server/src/main/java/org/opensearch/search/sandboxing/tracker/SandboxResourceUsageTrackerService.java

```diff
@@ -0,0 +1,108 @@
+ * This class tracks requests per Sandbox
+ */
+public class SandboxResourceUsageTrackerService implements
+    SandboxResourceUsageTracker,
+        TaskManager.TaskEventListeners {
+
+    public static final List<SandboxResourceType> TRACKED_RESOURCES = List.of(SandboxResourceType.JVM);
+
+    private final LongSupplier timeNanosSupplier;
+    private final TaskManager taskManager;
+    private final TaskResourceTrackingService taskResourceTrackingService;
+
+    /**
+     * SandboxResourceTrackerService constructor
+     * @param taskManager
+     * @param taskResourceTrackingService
+     * @param timeNanosSupplier
+     */
+    @Inject
+    public SandboxResourceUsageTrackerService(
+        final TaskManager taskManager,
+        final TaskResourceTrackingService taskResourceTrackingService,
+        final LongSupplier timeNanosSupplier
+    ) {
+        this.taskManager = taskManager;
+        this.taskResourceTrackingService = taskResourceTrackingService;
+        this.timeNanosSupplier = timeNanosSupplier;
+    }
+
+    @Override
+    public SandboxResourceTaskComposite getSandboxResourceTaskComposite() {
+        Map<String, List<Task>> tasksBySandbox = getTasksGroupedBySandbox();
+        Map<String, Map<SandboxResourceType, Long>> sandboxResourceUsage = getResourceUsageOfSandboxes(tasksBySandbox);
+
+        TaskData taskData = new TaskData(tasksBySandbox);
+        ResourceUsageData resourceUsageData = new ResourceUsageData(sandboxResourceUsage);
+        return new SandboxResourceTaskComposite(resourceUsageData, taskData);
+    }
+
+    private Map<String, List<Task>> getTasksGroupedBySandbox() {
+        return taskResourceTrackingService.getResourceAwareTasks()
+            .values()
+            .stream()
+            .filter(SandboxTask.class::isInstance)
+            .map(SandboxTask.class::cast)
+            .collect(Collectors.groupingBy(
+                SandboxTask::getSandboxName,
+                Collectors.mapping(task -> (Task) task, Collectors.toList())
+            ));
+    }
+
+    private Map<String, Map<SandboxResourceType, Long>> getResourceUsageOfSandboxes(Map<String, List<Task>> tasksBySandbox) {
+        Map<String, Map<SandboxResourceType, Long>> resourceUsage = new HashMap<>();
+
+        // Iterate over each sandbox entry
+        for (Map.Entry<String, List<Task>> sandboxEntry : tasksBySandbox.entrySet()) {
+            String sandboxName = sandboxEntry.getKey();
+            List<Task> tasks = sandboxEntry.getValue();
+
+            // Prepare a usage map for the current sandbox, or retrieve the existing one
+            Map<SandboxResourceType, Long> sandboxUsage = resourceUsage.computeIfAbsent(sandboxName, k -> new HashMap<>());
+
+            // Accumulate resource usage for each task in the sandbox
+            for (Task task : tasks) {
+                for (SandboxResourceType resourceType: TRACKED_RESOURCES) {
+                    sandboxUsage.put(resourceType, resourceType.getResourceUsage(task));
```

Comment:
This logic is incorrect and the sandbox usage for a sandbox will only hold the resource usage for the last visited task.